library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionActivity.kt

```diff
@@ -103,16 +98,30 @@ internal class TransactionActivity : BaseChuckerActivity() {
         }
     }
 
-    private fun setupViewPager(viewPager: ViewPager) {
-        adapter = TransactionPagerAdapter(this, supportFragmentManager)
-        viewPager.adapter = adapter
-        viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
-            override fun onPageSelected(position: Int) {
-                selectedTabPosition = position
-                populateUI(transaction)
+    private fun setupViewPager() {
+        val viewPager = findViewById<ViewPager2>(R.id.viewPagerTransaction)
+
+        viewPagerAdapter = TransactionPagerAdapter(this)
+        viewPager.apply {
+            adapter = viewPagerAdapter
+            offscreenPageLimit = TransactionPagerAdapter.TRANSACTION_SCREEN_OFFSCREEN_LIMIT as Int
+            registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+                override fun onPageSelected(position: Int) {
+                    selectedTabPosition = position
+                    populateUI(transaction)
+                }
+            })
+            currentItem = selectedTabPosition
+        }
+
+        val tabLayout = findViewById<TabLayout>(R.id.tabLayoutTransaction)
```

Comment:
```suggestion
        val tabLayout : TabLayout = findViewById(R.id.tabLayoutTransaction)
```

Reply:
I think that we should drop all these `findViewById` usages. But in another PR later.

Reply:
Yup, feel free to drop it for now.